lisa/wlgen/rta.py

```diff
@@ -408,15 +410,16 @@ def _calibrate(cls, target, res_dir):
 
         logger.info('Target RT-App calibration: %s', pload)
 
-        if 'sched' not in target.modules:
-            return pload
+        plat_info = target.plat_info
 
         # Sanity check calibration values for asymmetric systems
-        cpu_capacities = target.sched.get_capacities()
+        cpu_capacities = plat_info['cpu-capacities']
 
-        # Find the max pload per capacity level
+        # Find the min pload per capacity level, i.e. the fastest detected CPU.
+        # It is more likely to represent the right pload, as it has suffered
+        # from less IRQ slowdown or similar disturbances that might be random.
         capa_pload = {
-            capacity: max(pload[cpu] for cpu, capa in cpu_caps)
+            capacity: min(pload[cpu] for cpu, capa in cpu_caps)
```

Comment:
The reason for using max watch to catch e.g. a rogue big CPU that happened to have a bigger pload than a LITTLE. So you might want to keep it there (or do both).

Reply:
problem with max() is that we select the pload that is the less relevant for computing the capacity. Background noise will always lead to higher pload, so the minimum one is the most representative.

I'll add another check to find the condition you mentioned instead.